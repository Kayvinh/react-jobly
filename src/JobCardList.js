import React from 'react';
import JobCard from './JobCard';


/** Renders a list of jobs 
 * 
 * Props
 * -jobs: [{id, title, salary, equity, companyName}, ...]
 * 
 * State
 * -none
 * 
 * {CompanyDetail, JobList} -> JobCardList -> JobCard
*/

function JobCardList({ jobs }) {
  return (
    <div className='JobCardList'>
      {jobs.map(j => 
        <JobCard 
          key={j.id} 
          title={j.title} 
          companyName={j.companyName}  
          salary={j.salary}
          equity={j.equity}
        />)}
    </div>
  );
}

export default JobCardList;