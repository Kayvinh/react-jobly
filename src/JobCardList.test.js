import React from 'react';
import { render } from '@testing-library/react';
import JobCardList from './JobCardList';



const testJobCardList = <JobCardList jobs={[
    {
        id: 1,
        title: "test title 1",
        companyName: "test company name 1",
        salary: 100,
        equity: 0.01
    },
    {
        id: 2,
        title: "test title 2",
        companyName: "test company name 2",
        salary: 200,
        equity: 0.02
    },
    {
        id: 3,
        title: "test title 3",
        companyName: "test company name 3",
        salary: 300,
        equity: 0.03
    }
]} />;


test('renders without crashing', function () {
    render(testJobCardList);
});

test('renders all jobCards', function () {
    const { container } = render(testJobCardList);

    expect(container).toContainHTML("test title 1");
    expect(container).toContainHTML("test title 2");
    expect(container).toContainHTML("test title 3");
});

test('matches snapshot', function () {
    const { container } = render(testJobCardList);

    expect(container).toMatchSnapshot()

});