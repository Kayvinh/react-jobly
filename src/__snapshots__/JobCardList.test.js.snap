// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="JobCardList"
  >
    <div
      class="JobCard card"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title"
        >
          <h6>
            <b>
              test title 1
            </b>
            <p>
              test company name 1
            </p>
          </h6>
        </div>
        <small>
          <div>
            Salary: 
            100
          </div>
          <div>
            Equity: 
            0.01
          </div>
        </small>
      </div>
    </div>
    <div
      class="JobCard card"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title"
        >
          <h6>
            <b>
              test title 2
            </b>
            <p>
              test company name 2
            </p>
          </h6>
        </div>
        <small>
          <div>
            Salary: 
            200
          </div>
          <div>
            Equity: 
            0.02
          </div>
        </small>
      </div>
    </div>
    <div
      class="JobCard card"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title"
        >
          <h6>
            <b>
              test title 3
            </b>
            <p>
              test company name 3
            </p>
          </h6>
        </div>
        <small>
          <div>
            Salary: 
            300
          </div>
          <div>
            Equity: 
            0.03
          </div>
        </small>
      </div>
    </div>
  </div>
</div>
`;
